aten/src/ATen/native/cpu/DistributionTemplates.h

```diff
@@ -266,12 +275,14 @@ struct LogNormalKernel {
 // =================================================== Geometric ======================================================
 
 template<typename RNG>
-void geometric_kernel(TensorIterator& iter, double p, RNG generator) {
-  AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "geometric_cpu", [&]() {
+void geometric_kernel(TensorIterator& iter, double p_, RNG generator) {
+  AT_DISPATCH_ALL_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, iter.dtype(), "geometric_cpu", [&]() {
+    using accscalar_t = at::GeometricType<scalar_t>::type;
```

Comment:
This doesn't look semantics preserving? Previously we unconditionally created a geometric distribution on `double`, and now you vary based on `scalar_t`. Intentional?

Reply:
Absolutely intentional, because we need `double` precision only for dtype `double`. What's the point of calculating in `double` and cast the result to less precise dtype?

Reply:
I mean, it depends on the case! For example, we do accumulations in double and then cast to float because too much inaccuracy may accumulate over successive FP computations if done in float. I don't know what computation geometric distribution is doing, but it's in *general* not sound reduce the precision of internal computations. Now, I could certainly see that in this specific case it might not matter, but you should be making an argument here about the specifics of the computation in question.

Reply:
Understand, agree. I reverted all CPU's kernels to `double` and added `scalar_t` changes in subsequent PRs.
But! Why do we use `scalar_t`/`acctype` in CUDA kernels but `double` in CPU's? This inconsistency worries me.

Reply:
The inconsistency could definitely be a bug. All I ask is for the bug to be resolved in a separate PR.